backend/src/utils/paletteResponseFactories.ts

```diff
@@ -0,0 +1,39 @@
+import { PaletteAPIErrorData, PaletteAPIResponse } from "palette-types";
+
+/**
+ * Creates a success response for the Palette API.
+ * @param data The data to include in the response.
+ * @param message Optional message to include in the response.
+ * @returns A PaletteAPIResponse object indicating success.
+ */
+export function newPaletteSuccessResponse<T>(
+  data: T,
+  message?: string,
+): PaletteAPIResponse<T> {
+  return {
+    data,
+    success: true,
+    message: message || "Request successful",
+    error: null,
+    errors: [],
+  };
+}
+
+/**
+ * Creates an error response for the Palette API.
+ * @param error The error message to include in the response.
+ * @param errors Optional array of PaletteAPIErrorData to include in the response.
+ * @returns A PaletteAPIResponse object indicating failure.
+ */
+export function newPaletteErrorResponse(
```

Comment:
just for naming consistency with other factory functions, can we update these to `createPaletteSuccessResponse` and `createPaletteErrorResponse`